app/modules/entities/components/layouts/works_browser_section.svelte

```diff
@@ -1,50 +1,131 @@
   import WorkGridCard from '#entities/components/layouts/work_grid_card.svelte'
-  import { i18n, I18n } from '#user/lib/i18n'
+  import WorkActions from '#entities/components/layouts/work_actions.svelte'
+  import { addWorksImages } from '#entities/lib/types/work_alt'
   import { bySearchMatchScore, getSelectedUris } from '#entities/components/lib/works_browser_helpers'
-  import { onChange } from '#lib/svelte/svelte'
   import { flip } from 'svelte/animate'
+  import { i18n } from '#user/lib/i18n'
+  import { onChange } from '#lib/svelte/svelte'
   import { setIntersection } from '#lib/utils'
+  import { screen } from '#lib/components/stores/screen'
+  import Flash from '#lib/components/flash.svelte'
 
   export let section, displayMode, facets, facetsSelectedValues, textFilterUris
 
-  const { label, entities: works } = section
+  const { entities: works } = section
+  let { label, context } = section
+
+  let filteredWorks = works
+  let paginatedWorks = []
+  let flash
+  if (context) {
+    flash = {
+      type: 'warning',
+      message: context
+    }
+  }
 
-  // TODO: display only the first n items, and add more on scroll
-  let displayedWorks = works
   function filterWorks () {
     if (!facetsSelectedValues) return
     let selectedUris = getSelectedUris({ works, facets, facetsSelectedValues })
     if (textFilterUris) selectedUris = setIntersection(selectedUris, textFilterUris)
-    displayedWorks = works.filter(work => selectedUris.has(work.uri))
+    filteredWorks = works.filter(filterSelectedWorks(selectedUris, facetsSelectedValues))
     if (textFilterUris) {
-      displayedWorks = displayedWorks.sort(bySearchMatchScore(textFilterUris))
+      filteredWorks = filteredWorks.sort(bySearchMatchScore(textFilterUris))
     }
   }
 
+  const filterSelectedWorks = (selectedUris, facetsSelectedValues) => work => {
+    const { uri } = work
+    return selectedUris.has(uri) || isSelectedEntityAParent(uri, facetsSelectedValues)
+  }
+
+  const isSelectedEntityAParent = (uri, facetsSelectedValues) => {
+    // ie. if a collection is selected, display the collection in question.
+    // TODO: generalize pattern to series
+    return facetsSelectedValues['wdt:P195'] === uri
+  }
+
   $: onChange(facetsSelectedValues, textFilterUris, filterWorks)
+
+  function onWorksScroll (e) {
+    const { scrollTop, scrollTopMax } = e.currentTarget
+    if (scrollTopMax < 100) return
+    if (scrollTop + 100 > scrollTopMax) displayLimit += 10
+  }
+
+  // Limit needs to be high enough to have enough elements in order to be scrollable
+  // otherwise on:scroll wont be triggered
+  let initialLimit = 25
+  let displayLimit = initialLimit
+
+  let scrollableElement
+
+  async function addMoreWorks () {
+    paginatedWorks = filteredWorks.slice(0, displayLimit)
+    await addImagesToPaginatedWorks()
+  }
+
+  async function addImagesToPaginatedWorks () {
+    paginatedWorks = await addWorksImages(paginatedWorks)
```

Comment:
Calling `addWorksImages` on all `paginatedWorks` everytime `addMoreWorks` is called leads to making a lot of duplicated requests, see the kind of server logs I get
```
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 281ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 235ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 142ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 221ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 197ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 304 216ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 204ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 205ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 220ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 191ms
GET /api/entities?action=images&uris=inv:e6a1a3d983761e6883db00ee351314ab|inv:f38d6bb78ae4af173a3a9708d790b11b|inv:f911a92a59231042b4e92f81b67ed506|inv:f911a92a59231042b4e92f81b6e600ee|inv:fc498b96cf319a9c5e1fa9209156ca81|wd:Q30958732|wd:Q30958741|wd:Q30958747|wd:Q30963508|wd:Q30963518|wd:Q98831221|wd:Q110270091|wd:Q110278997|wd:Q30958629|wd:Q30958635|wd:Q30958644|wd:Q30958731|wd:Q7742168|wd:Q7757806|wd:Q9632979 200 191ms
```
In this case, the answers are pretty fast, because those entities images were already cached, but in case of cache miss, all those requests will trigger duplicated calls to multiple external services (Wikipedia, OpenLibrary, & co).

Some work around ideas: either make sure that no work is passed twice to `addWorksImages` (passing only the newly displayed works for instance), or change `addWorksImages` internals to let it know images for a given entity is already being fetched (seems to be a more dirty option)

Reply:
> I modified the limit to 200 in aa124b7

Are we sure about 200 ? Its kind of low, as some prolific authors dont have all their works displayed (ie. `/entity/wd:Q34981`, I. Asimov). 500 sounds more adequate? Another possibility would be to limit by entityType (ie. 200 for publishers, subjects, genre and no limit for the rest) (?)

Would

Reply:
> make sure that no work is passed twice to addWorksImages (passing only the newly displayed works for instance)

done this way in 16a8e22060f97c5c904f498aaee1aeed37c392da at the cost of an extra filtering though, which is necessary because of facetFiltering adding randomly some new works

Reply:
related commit : 916bd03d48cb86ea4ffa9842c62d9dc6afecbb4b
which tries to make it faster by not waiting for images, at the cost of re-assinging the whole entities list (maybe not worth it? [droppable commit])